snapcraft/plugins/jhbuild.py

```diff
@@ -0,0 +1,773 @@
+
+            self.run([
+                'git',
+                'apply',
+            ], cwd=os.path.join(
+                self.sourcedir,
+                'jhbuild',
+            ), input=ROOT_PATCH)
+
+    def _setup_jhbuild(self):
+        LOG.info('Building JHBuild')
+
+        if os.path.isdir(self.builddir):
+            if shutil.rmtree.avoids_symlink_attacks:
+                shutil.rmtree(self.builddir)
+
+        shutil.copytree(os.path.join(self.sourcedir, 'jhbuild'), self.builddir)
+
+        env = {
+            'CCACHE_DIR': (
+                self.options.ccache_cache or
+                os.path.join(self.builddir, '_ccache')
+            ),
+        }
+
+        self.run([
+            './autogen.sh',
+            '--prefix=%s' % os.path.join(
+                self.installdir, 'usr',
+            ),
+        ], env=env, cwd=self.builddir)
+
+        self.run([
+            'make',
+            '-j%d' % self.parallel_build_count,
+        ], env=env, cwd=self.builddir)
+
+        self.run([
+            'make',
+            '-j%d' % self.parallel_build_count,
+            'install',
+        ], env=env, cwd=self.builddir)
+
+        self.jhbuild_program = os.path.join(
+            self.installdir, 'usr', 'bin', 'jhbuild',
+        )
+
+    def write_jhbuildrc(self):
+        """
+        write_jhbuildrc() - write out the jhbuildrc file for build dependency
+        specification
+        """
+        os.makedirs(os.path.join(self.installdir, 'etc'), exist_ok=True)
+
+        with open(self.jhbuildrc_path, 'w') as jhbuildrc_file:
+            jhbuildrc_file.write('moduleset = \'%s\'\n' %
+                                 self.options.module_set)
+            if self.options.module_set_dir:
+                jhbuildrc_file.write('modulesets_dir = \'%s\'\n' % (
+                    self.options.module_set_dir
+                ))
+            jhbuildrc_file.write('tarballdir = \'%s\'\n' %
+                                 (self.options.jhbuild_archive or
+                                  os.path.join(self.partdir,
+                                               'jhbuild_archive')))
+            jhbuildrc_file.write('dvcs_mirror_dir = \'%s\'\n' %
+                                 os.path.join(self.partdir, 'jhbuild_mirror'))
+            jhbuildrc_file.write('checkoutroot = \'%s\'\n' % self.sourcedir)
+            jhbuildrc_file.write('buildroot = \'%s\'\n' % self.builddir)
+            jhbuildrc_file.write('prefix = \'%s\'\n' %
+                                 os.path.join(self.installdir, 'usr'))
+            jhbuildrc_file.write('use_local_modulesets = True\n')
+            jhbuildrc_file.write('skip = [%s]\n' % ', '.join([
+                '\'%s\'' % module
+                for module in self.skip_modules
+            ]))
+            jhbuildrc_file.write(
+                'module_autogenargs[\'gdk-pixbuf\'] = '
+                '\'--disable-gio-sniffing\'\n'
+            )
```

Comment:
This might be a bit easier if you specify the whole configuration in a single go, and use the `str.format()` method to substitute in named parameters.  Something like:

    config = '''
    moduleset = {module_set!r}
    ...
    '''
    jhbuildrc_file.write(config.format(module_set=self.options.module_set, ...))